### PR TITLE
Feat/erc20

### PR DESCRIPTION
**Pull request by xvc888**
première tentative de pull request

### CODE REVIEW

File: truffle/contracts/Byx.sol [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// Byx.sol
+
+pragma solidity 0.8.14;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract Byx is ERC20 {
+    constructor() ERC20("Byx Stablecoin", "BYX") {}
+
+    // fonction faucet pour créer des Dai tokens
```
Comment (0xB19) [inline truffle/contracts/Byx.sol]: Changer nom du token dans le commmentaire

File: truffle/contracts/MyByxProject.sol [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+// MyByxProject.sol
+
+pragma solidity 0.8.14;
+import "../node_modules/@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract MyByxProject {
+    IERC20 byx;
+
+    constructor(address byxAddress) {
```
Comment (0xB19) [inline truffle/contracts/MyByxProject.sol]: Dans les contrats, précéder le nom des params par un underscore ex: address _byxAddress

File: truffle/contracts/Byx.sol (hunk 1 above)
Comment (xvc888) [reply truffle/contracts/Byx.sol]: Done! Merci!

File: truffle/contracts/MyByxProject.sol (hunk 2 above)
Comment (xvc888) [reply truffle/contracts/MyByxProject.sol]: Done!

**Review by YvesToiser — COMMENTED** [review]
client/package.json

hdwallet-provider & dotenv do not belong here (only in truffle/package.json)

**Review by YvesToiser — COMMENTED** [review]
- No need for VRF contract

**Review by YvesToiser — COMMENTED** [review]
truffle/migrations/1_deploy_simple_storage.js

should rename contract
actually delete it as we do not need this contract
then you should rename the other deployer as n°1

**Review by YvesToiser — COMMENTED** [review]
truffle/migrations/2_deploy_contracts.js

you could add some comments to explain deployment process

**Review by YvesToiser — COMMENTED** [review]
truffle/contracts/Byx.sol

Byx is not a stablecoin!

**Review by YvesToiser — COMMENTED** [review]
truffle/contracts/MyByxProject.sol

"foo" is a name for example. Never use it in a real project. name de function depending on what it does

**Review by YvesToiser — COMMENTED** [review]
should review comments of all contracts
- Adapt them to project
- Put them in english
- Use natspec format

**Pull request merged**